stdlib/SparseArrays/test/sparsevector.jl

```diff
@@ -713,6 +713,15 @@ end
     @test sum(x) == 4.0
     @test sum(abs, x) == 5.5
     @test sum(abs2, x) == 14.375
+    @test @inferred(sum(t -> true, x)) === 8
+    @test @inferred(sum(t -> abs(t) + one(t), x)) == 13.5
+
+    @test @inferred(sum(t -> true, spzeros(Float64, 8))) === 8
+    @test @inferred(sum(t -> abs(t) + one(t), spzeros(Float64, 8))) === 8.0
+
+    @test_broken sum(t -> true, spzeros(Float64, 0)) === 0 # reducing over an empty collection
```

Comment:
I know it's unrelated, but maybe throw an `@test_broken sum(t -> true, zeros(Float64, 0)) === 0` right next to this line so that way when one gets fixed the author will find the other.